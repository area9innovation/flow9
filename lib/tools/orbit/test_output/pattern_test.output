Flow compiler (3rd generation)

Processing 'orbit' on http server

A 1b
Result 1: It's one
Result 2: 2
Result 3: 3
Result 4: 4
Found person named Alice who is 30 years old
Result: "Found person named Alice who is 30 years old"
