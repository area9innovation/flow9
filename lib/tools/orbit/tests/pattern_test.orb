// Test file for multiple pattern matching

// Define a function that tests pattern matching with multiple patterns
fn test_match(x) = 
  match x {
    1 => "It's one";
    2 => "It's two";
    3 => "It's three";
    y => "It's something else: " + y
  };

// Test with different values
let result1 = test_match(1);
println("Result 1: " + result1);

let result2 = test_match(2);
println("Result 2: " + result2);

let result3 = test_match(3);
println("Result 3: " + result3);

let result4 = test_match(4);
println("Result 4: " + result4);

// Test with constructors
let person = Person("Alice", 30);

match person {
  Person(name, age) => println("Found person named " + name + " who is " + age + " years old");
  Other() => println("Not a person")
}