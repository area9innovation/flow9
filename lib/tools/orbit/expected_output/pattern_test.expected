A 1b
Result 1: It's one
Result 2: It's two
Result 3: It's three
Result 4: It's something else: 4
Found person named Alice who is 30 years old
Result: 0
